offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/product/MaintainProductsView.groovy

```diff
@@ -157,14 +163,38 @@ class MaintainProductsView extends VerticalLayout{
             createProductView.setVisible(false)
         })
 
+        copyProduct.addClickListener({
+            viewModel.productUpdate.emit(viewModel.selectedProduct)
+            maintenanceLayout.setVisible(false)
+            copyProductView.setVisible(true)
+        })
+
+        copyProductView.abortButton.addClickListener({
+            maintenanceLayout.setVisible(true)
+            copyProductView.setVisible(false)
+        })
+
         archiveProduct.addClickListener({
             controller.archiveProduct(viewModel.selectedProduct.productId)
         })
 
         viewModel.products.addPropertyChangeListener({
             productGrid.dataProvider.refreshAll()
         })
+        viewModel.addPropertyChangeListener({
```

Comment:
shouldn't this changelistener be on the grid where you select something?

Reply:
You can either do a call on the grid with a selection listener (add the function you call to the upper listener) or you can check for the change of the specific property (selectedProduct) in the viewmodel to be more concrete

Reply:
Good Idea 👍